src/uproot/interpretation/objects.py

```diff
@@ -281,8 +300,10 @@ def _discover_forth(self, data, byte_offsets, branch, context, cursor_offset):
             cursor = uproot.source.cursor.Cursor(
                 0, origin=-(byte_start + cursor_offset)
             )
+
             if "forth" in context.keys():
-                context["forth"].gen.var_set = False
+                context["forth"].gen.count_obj = 0
```

Comment:
This is an unrelated fix. I noticed that when we do get to the next entry, the `count_obj` keeps increasing so `ForthGenerator.get_keys` returns numbers for keys that don't exist. For instance, the keys are

```python
["node0-offsets", "node1-offsets", "node2-data"]
```

and `get_keys` was returning `0` and `1` for offsets in the first entry, `3` and `4` in the second entry, etc. I could see that the integer was being used specifically to build references to these keys in the AwkwardForth code, so I'm quite certain that this is the right thing to do.

But if it doesn't look right to you, let me know.